src/main/java/net/devtech/grossfabrichacks/field/FieldSynthesizer.java

```diff
@@ -0,0 +1,133 @@
+package net.devtech.grossfabrichacks.field;
+
+import java.util.List;
+import net.devtech.grossfabrichacks.transformer.TransformerApi;
+import net.devtech.grossfabrichacks.util.ASMUtil;
+import net.devtech.grossfabrichacks.util.DelegatingInsnList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.MethodNode;
+
+public class FieldSynthesizer {
+    private static final Logger LOGGER = LogManager.getLogger("GrossFabricHacks/FieldSynthesizer");
+
+    public static void init() {
+        LOGGER.info("Initializing field synthesis capability.");
+
+        TransformerApi.registerPreMixinAsmClassTransformer(FieldSynthesizer::transform);
+
+        LOGGER.info("Initialization complete. Multiple inheritance of state is now at your key presses.");
+    }
+
+    private static void transform(final String name, final ClassNode klass) {
+        if (klass.visibleAnnotations != null) {
+            for (final AnnotationNode annotation : klass.visibleAnnotations) {
+                if ("Lnet/devtech/grossfabrichacks/field/Fields;".equals(annotation.desc)) {
+                    //noinspection unchecked
+                    for (final AnnotationNode field : (List<AnnotationNode>) ASMUtil.getAnnotationValue(annotation, "value")) {
+                        addField(klass, field);
+                    }
+                }
+            }
+        }
+
+        for (final MethodNode method : klass.methods) {
+            if (method.visibleAnnotations != null) {
+                for (final AnnotationNode annotation : method.visibleAnnotations) {
+                    if (Type.getDescriptor(Getter.class).equals(annotation.desc)) {
+                        FieldSynthesizer.get(name, method, annotation);
+                    } else if (Type.getDescriptor(Setter.class).equals(annotation.desc)) {
+                        FieldSynthesizer.set(name, method, annotation);
```

Comment:
like here, why can't u just add fields to the class, u have the information required, so why not use it

Reply:
I could do that; it's what I did before—but it turned out badly—so I came up with the present `@Fields.Entry` declaration. It also more closely resembles typical Java structure than generating fields from accessors (not to suggest that fields declared via annotations resemble typical Java code in any way).